packages/playground/blueprints/src/lib/steps/set-site-language.ts

```diff
@@ -0,0 +1,155 @@
+			require '${docroot}/wp-includes/version.php';
+			echo $wp_version;
+		`,
+		})
+	).text;
+
+	const translations = [
+		{
+			url: `https://downloads.wordpress.org/translation/core/${wpVersion}/${language}.zip`,
+			type: 'core',
+		},
+	];
+
+	const pluginListResponse = await playground.run({
+		code: `<?php
+		require_once('${docroot}/wp-load.php');
+		require_once('${docroot}/wp-admin/includes/plugin.php');
+		echo json_encode(
+			array_values(
+				array_map(
+					function($plugin) {
+						return [
+							'slug'    => $plugin['TextDomain'],
+							'version' => $plugin['Version']
+						];
+					},
+					array_filter(
+						get_plugins(),
+						function($plugin) {
+							return !empty($plugin['TextDomain']);
+						}
+					)
+				)
+			)
+		);`,
+	});
+
+	const plugins = pluginListResponse.json;
+	for (const { slug, version } of plugins) {
+		translations.push({
+			url: `https://downloads.wordpress.org/translation/plugin/${slug}/${version}/${language}.zip`,
+			type: 'plugin',
+		});
+	}
+
+	const themeListResponse = await playground.run({
+		code: `<?php
+		require_once('${docroot}/wp-load.php');
+		require_once('${docroot}/wp-admin/includes/theme.php');
+		echo json_encode(
+			array_values(
+				array_map(
+					function($theme) {
+						return [
+							'slug'    => $theme->get('TextDomain'),
+							'version' => $theme->get('Version')
+						];
+					},
+					wp_get_themes()
+				)
+			)
+		);`,
+	});
+
+	const themes = themeListResponse.json;
+	for (const { slug, version } of themes) {
+		translations.push({
+			url: `https://downloads.wordpress.org/translation/theme/${slug}/${version}/${language}.zip`,
+			type: 'theme',
+		});
+	}
+
+	if (!(await playground.isDir(`${docroot}/wp-content/languages/plugins`))) {
+		await playground.mkdir(`${docroot}/wp-content/languages/plugins`);
+	}
+	if (!(await playground.isDir(`${docroot}/wp-content/languages/themes`))) {
+		await playground.mkdir(`${docroot}/wp-content/languages/themes`);
+	}
+
+	for (const { url, type } of translations) {
+		try {
+			const response = await fetch(url);
+			if (!response.ok) {
+				throw new Error(
+					`Failed to download translations for ${type}: ${response.statusText}`
+				);
+			}
+
+			let destination = `${docroot}/wp-content/languages`;
+			if (type === 'plugin') {
+				destination += '/plugins';
+			} else if (type === 'theme') {
+				destination += '/themes';
+			}
+
+			await unzipFile(
+				playground,
+				new File([await response.blob()], `${language}-${type}.zip`),
+				destination
+			);
+		} catch (error) {
+			/**
+			 * If a core translation wasn't found we should throw an error because it means the language is not supported or the language code isn't correct.
+			 */
+			if (type === 'core') {
+				throw new Error(
+					`Failed to download translations for WordPress. Please check if the language code ${language} is correct. You can find all available languages and translations on https://translate.wordpress.org/.`
+				);
+			}
```

Comment:
I couldn't find a way in WP to check if the language code is valid, so I'm relying on API errors.
If a translation for WordPress core doesn't exist, the step will fail.

Theme and plugin translations are still allowed to fail because they have much lower translation rates.